ros2doctor/ros2doctor/api/format.py

```diff
@@ -44,33 +46,13 @@ def compute_padding(report_items: List[Tuple[str, str]]) -> int:
     return padding
 
 
-def custom_warning_format(msg, cat, filename, linenum, file=None, line=None):
-    return '%s: %s: %s: %s\n' % (filename, linenum, cat.__name__, msg)
-
-
-class CustomWarningFormat:
-    """Support custom warning format without modifying default format."""
-
-    def __enter__(self):
-        self._default_format = warnings.formatwarning
-        warnings.formatwarning = custom_warning_format
-
-    def __exit__(self, t, v, trb):
-        """
-        Define exit action for context manager.
-
-        :param t: type
-        :param v: value
-        :param trb: traceback
-        """
-        warnings.formatwarning = self._default_format
-
-
-def doctor_warn(msg: str) -> None:
+def doctor_warn() -> Callable:
     """
-    Use CustomWarningFormat to print customized warning message.
+    Print customized warning message with package and line info.
 
     :param msg: warning message to be printed
     """
-    with CustomWarningFormat():
-        warnings.warn(msg)
+    def custom_warning_format(message, category, filename, lineno, file=None, line=None):
+        return f'{filename}:{lineno}: {category.__name__}: {message}\n'
+    warnings.formatwarning = custom_warning_format
+    return warnings.warn
```

Comment:
The `doctor_warn()('...')` looks a little bit weird to me. As an alternative, I think we can get rid of the extra parenthesis and make the message have the right file name by using `stacklevel`: https://docs.python.org/3/library/warnings.html#warnings.warn

```python3
with CustomWarningFormat():
    warnings.warn(msg, stacklevel=2)
```